src/eventloop.c

```diff
@@ -241,45 +161,69 @@ void handle_input(uv_poll_t* handle, int status, int events) {
 */
 guint input_add(int fd, PurpleInputCondition cond,
                 PurpleInputFunction func, gpointer user_data) {
+    g_return_if_fail(cond == 1 || cond == 2);
```

Comment:
If this is the same "magic number" as above, then it seems like we should be able to use `PURPLE_INPUT_READ` and `PURPLE_INPUT_WRITE` – but since these are the only two values according to `libpurple/eventloop.h`, then what's going on here? Is this something else in `PurpleInputCondition`'s clothing? Seems like it, given the assignment to `->events` :) Could you add some clarification for that?

Reply:
I was trying to be clever here and validate our inputs, but it might be excessive given I guess we can expect the caller to be doing that? And yeah this is me mapping `PurpleInputCondition` to `uv_poll_event`